mapping/mapping.go

```diff
@@ -55,4 +55,6 @@ type IndexMapping interface {
 
 	AnalyzerNameForPath(path string) string
 	AnalyzerNamed(name string) analysis.Analyzer
+
+	FieldMappingForPath(path string) *FieldMapping
```

Comment:
perhaps its better to not return a pointer to the fieldMapping. mainly because i think this interface deals with reading of the content in the index mapping, and passing a pointer to the fieldMapping could lead to users having the option to modify the config of the field mapping at instances when they aren't allowed to do so (for example query time).